lark/visitors.py

```diff
@@ -158,7 +158,11 @@ def _transform_tree(self, tree):
 
     def transform(self, tree: Tree[_Leaf_T]) -> _Return_T:
         "Transform the given tree, and return the final result"
-        return self._transform_tree(tree)
+        res = list(self._transform_children([tree]))
```

Comment:
this has the side effect of also allowing everything that isn't a tree or token to be passed without error. Not sure if that is good behavior.

Reply:
What's bad about it?

Reply:
It might hide user error where they pass things that aren't the result of a `Lark.parse` call, which is probably never intentional. But this isn't a strong objection, if you think this is fine, go ahead.

Reply:
I see your point, but I think I'll keep it. My reasoning is that:

1) Lark.parse may return an arbitrary object (through another transformer)

2) It would make the root behave the same as the leaf, in which we allow arbitrary objects and just skip them without an error

3) It's not the kind of mistake that is likely to proceed much without being noticed down the line. (though I admit in some edge cases it might happen)